Documentation/troubleshooting.md

```diff
@@ -1,4 +1,84 @@
-### Firewalls
+# Troubleshooting
+
+# General
+
+## Logging
+Flannel uses the `glog` library but only supports logging to stderr. The severity level can't be changed but the verbosity can be changed with the `-v` option. flannel does not make extensive use of the verbosity level but increasing th value from `0` (the default) will result in some additional logs. To get the most detailed logs, use `-v=10`
+
+```
+-v value
+    	log level for V logs
+-vmodule value
+    	comma-separated list of pattern=N settings for file-filtered logging
+-log_backtrace_at value
+    	when logging hits line file:N, emit a stack trace
+```
+
+When running under systemd (e.g. on CoreOS container linux) the logs can be viewed with `journalctl -u flanneld`
+
+When flannel is running as a pod on Kubernetes, the logs can be viewed with `kubectl logs --namespace kube-system <POD_ID> -c kube-flannel`. You can find the pod IDs with `kubectl get po --namespace kube-system -l app=flannel`
+
+## Interface selection and the public IP.
+Most backends require that each node has a unique "public IP" address. This address is chosen when flannel starts, and if the address changes then flannel needs to be restarted. Since leases are tied to the public address
+
+The interface chosen and the public IP in use is logged out during startup, e.g.
+```
+I0629 14:28:35.866793    5522 main.go:386] Determining IP address of default interface
+I0629 14:28:35.866987    5522 main.go:399] Using interface with name enp62s0u1u2 and address 172.24.17.174
+I0629 14:28:35.867000    5522 main.go:412] Using 10.10.10.10 as external address
+```
+
+### Vagrant
+Running flannel inside vagrant can cause problems because by default all VMs will have two interfaces. The first is for external traffic that gets NATed. All hosts get the same IP address assigned to to this interface `10.0.2.15`. This causes problems because by default, flannel will select the first interface on a host which leads to all hosts thinking they have the same public IP address. The fix for this is to pass the `--iface eth1` flag to flannel so that the second interface is chosen.
+
```

Comment:
this paragraph is confusing, in that I'm not sure why all VMs will have two interfaces. is that just how they work (i don't think so)? or is it a result of the combination of flannel and Vagrant? Please make sure my suggestions are true before accepting them, cause i very well may be wrong.

Vagrant assigns two interfaces to all VMs. The first, for which all hosts are assigned the IP address `10.0.2.15`, is for external traffic that gets NATed. 

This may lead to problems with flannel. By default, flannel selects the first interface on a host. This leads to all hosts thinking they have the same public IP address. To prevent this issue, pass the `--iface eth1` flag to flannel so that the second interface is chosen.